### PR TITLE
Fix null pacakge

### PR DESCRIPTION
**Pull request by wuqian0808**
Package version is null when bizclassloader inited and hasn't any instance for this springapplication class.
Use startwith will throw null pointer exception
"Caused by: java.lang.NullPointerException
	at com.alipay.sofa.ark.springboot.listener.ArkApplicationStartListener.isSpringBoot2(ArkApplicationStartListener.java:82)"

### CODE REVIEW

**Comment by lvjing2** [comment]
> Package version is null when bizclassloader inited and hasn't any instance for this springapplication class. Use startwith will throw null pointer exception "Caused by: java.lang.NullPointerException at com.alipay.sofa.ark.springboot.listener.ArkApplicationStartListener.isSpringBoot2(ArkApplicationStartListener.java:82)"

Firstly, thanks for your PR. Please don't forget sign the [CLA](http://cla.sofastack.tech/). And could you show some demo to reproduction this problem?

**Comment by wuqian0808** [comment]
> > Package version is null when bizclassloader inited and hasn't any instance for this springapplication class. Use startwith will throw null pointer exception "Caused by: java.lang.NullPointerException at com.alipay.sofa.ark.springboot.listener.ArkApplicationStartListener.isSpringBoot2(ArkApplicationStartListener.java:82)"
> 
> Thanks for your PR. Please don't forget sign the [CLA](http://cla.sofastack.tech/), and could you show some demo to reproduction this problem?

cla scan passed.
Demo is here: https://github.com/wuqian0808/ark-demo
It's simple to recreate this issue by runnung main without then system var input: -Dsofa.ark.embed.enable=true...
Anyway there should be null predicate..

**Comment by zjulbj** [comment]
> > > Package version is null when bizclassloader inited and hasn't any instance for this springapplication class. Use startwith will throw null pointer exception "Caused by: java.lang.NullPointerException at com.alipay.sofa.ark.springboot.listener.ArkApplicationStartListener.isSpringBoot2(ArkApplicationStartListener.java:82)"
> > 
> > 
> > Thanks for your PR. Please don't forget sign the [CLA](http://cla.sofastack.tech/), and could you show some demo to reproduction this problem?
> 
> cla scan passed. Demo is here: https://github.com/wuqian0808/ark-demo It's simple to recreate this issue by runnung main without then system var input: -Dsofa.ark.embed.enable=true... Anyway there should be null predicate..

Thanks for you pr. But it seems the root cause is not the NullPointerException. The root cause is calling AbstractClasspathClassLoader.definePackageIfNecessary  when biz archive is DirectoryBizArchive

**Comment by wuqian0808** [comment]
> > > > Package version is null when bizclassloader inited and hasn't any instance for this springapplication class. Use startwith will throw null pointer exception "Caused by: java.lang.NullPointerException at com.alipay.sofa.ark.springboot.listener.ArkApplicationStartListener.isSpringBoot2(ArkApplicationStartListener.java:82)"
> > > 
> > > 
> > > Thanks for your PR. Please don't forget sign the [CLA](http://cla.sofastack.tech/), and could you show some demo to reproduction this problem?
> > 
> > 
> > cla scan passed. Demo is here: https://github.com/wuqian0808/ark-demo It's simple to recreate this issue by runnung main without then system var input: -Dsofa.ark.embed.enable=true... Anyway there should be null predicate..
> 
> Thanks for you pr. But it seems the root cause is not the NullPointerException. The root cause is calling AbstractClasspathClassLoader.definePackageIfNecessary when biz archive is DirectoryBizArchive. Could you fix it in BizFactoryServiceImpl.createBiz like :
> 
> ```
>   bizModel
>             .setClassLoader(
>                 new BizClassLoader(bizModel.getIdentity(), getBizUcp(bizModel.getClassPath()),
>                     bizArchive instanceof ExplodedBizArchive || bizArchive instanceof DirectoryBizArchive));
> ```

Added and all biz necessary classes get loaded. Tested fine.

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/service/biz/BizFactoryServiceImpl.java [hunk 1]
```diff
@@ -96,11 +96,10 @@ public Biz createBiz(BizArchive bizArchive) throws IOException {
             .setClassPath(bizArchive.getUrls());
 
         BizClassLoader bizClassLoader = new BizClassLoader(bizModel.getIdentity(),
-            getBizUcp(bizModel.getClassPath()), bizArchive instanceof ExplodedBizArchive);
+            getBizUcp(bizModel.getClassPath()), bizArchive instanceof ExplodedBizArchive || bizArchive instanceof DirectoryBizArchive);
```
Comment (zjulbj) [inline sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/service/biz/BizFactoryServiceImpl.java:99]: "import com.alipay.sofa.ark.loader.DirectoryBizArchive" is missing

Comment (wuqian0808) [reply sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/service/biz/BizFactoryServiceImpl.java:100]: Imported.. Sorry

**Comment by lvjing2** [comment]
@wuqian0808 Please format your code locally by executing the following command to pass the build.
`mvn clean install -DskipTests -Dmaven.javadoc.skip=true -B -U && sh ./check_format.sh`

**Comment by wuqian0808** [comment]
Use clean pr #513 instead. Previously pr I edited single file in chrome one by one so the format is incorrect

**Pull request closed without merging**